src/Orleans.Transactions.Azure/Storage/AzureTransactionLogStorage.cs

```diff
@@ -0,0 +1,364 @@
+                var commitRow = new CommitRow(nextLogSequenceNumber);
+
+                foreach (var transaction in transactionSegment)
+                {
+                    transaction.LSN = nextLogSequenceNumber++;
+                }
+
+                commitRow.Transactions = SerializeCommitRecords(transactionSegment);
+
+                batchOperation.Insert(commitRow);
+
+                if (batchOperation.Count == BatchOperationLimit)
+                {
+                    await table.ExecuteBatchAsync(batchOperation).ConfigureAwait(false);
+
+                    batchOperation = new TableBatchOperation();
+                }
+            }
+
+            if (batchOperation.Count > 0)
+            {
+                await table.ExecuteBatchAsync(batchOperation).ConfigureAwait(false);
+            }
+        }
+
+        public async Task TruncateLog(long lsn)
+        {
+            var continuationToken = default(TableContinuationToken);
+
+            var query = new TableQuery<CommitRow>().Where(TableQuery.CombineFilters(
+                TableQuery.GenerateFilterCondition(PartitionKey, QueryComparisons.Equal, CommitRecordPartitionKey),
+                TableOperators.And,
+                TableQuery.GenerateFilterCondition(RowKey, QueryComparisons.LessThanOrEqual, CommitRow.ToRowKey(lsn))));
+
+            do
+            {
+                var queryResult = await table.ExecuteQuerySegmentedAsync(query, continuationToken).ConfigureAwait(false);
+
+                continuationToken = queryResult.ContinuationToken;
+
+                if (queryResult.Results.Count > 0)
+                {
+                    var batchOperation = new TableBatchOperation();
+
+                    foreach (var row in queryResult)
+                    {
+                        var transactions = DeserializeCommitRecords(row.Transactions);
+
+                        if (transactions.Count > 0 && transactions[transactions.Count - 1].LSN <= lsn)
+                        {
+                            batchOperation.Delete(row);
+
+                            if (batchOperation.Count == BatchOperationLimit)
+                            {
+                                await table.ExecuteBatchAsync(batchOperation).ConfigureAwait(false);
+
+                                batchOperation = new TableBatchOperation();
+                            }
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+                }
+
+            } while (continuationToken != null);
+        }
+
+        public static Factory<Task<ITransactionLogStorage>> Create(IServiceProvider serviceProvider)
+        {
+            return async () =>
+            {
+                ObjectFactory factory = ActivatorUtilities.CreateFactory(typeof(AzureTransactionLogStorage), Type.EmptyTypes);
```

Comment:
In this case, a factory is being created, used, then discarded. Would `ActivatorUtilities.CreateInstance<AzureTransactionLogStorage>()` be a more appropriate alternative?

Reply:
yeah, this code is silly.  I refactored the pattern a couple of times and didn't really think through the final code well enough..